state-chain/chains/src/lib.rs

```diff
@@ -163,6 +167,53 @@ pub mod witness_period {
 		let floored_block_number = block_witness_floor(witness_period, block_number);
 		floored_block_number..=floored_block_number.saturating_add(witness_period - One::one())
 	}
+
+	impl<I: PartialOrd + Clone + Saturating> Step for BlockWitnessRange<I> {
+		fn steps_between(start: &Self, end: &Self) -> (usize, Option<usize>) {
+			// assert_eq!(start.period, end.period);
+			todo!()
+		}
+
+		fn forward_checked(mut start: Self, count: usize) -> Option<Self> {
+			(0..count)
+				.for_each(|_| start.root = start.root.clone().saturating_add(start.period.clone()));
+			Some(start)
+		}
+
+		fn backward_checked(mut start: Self, count: usize) -> Option<Self> {
+			(0..count)
+				.for_each(|_| start.root = start.root.clone().saturating_sub(start.period.clone()));
+			Some(start)
+		}
+	}
+
+	pub trait BlockZero {
```

Comment:
This was actually what I attempted to do first, but it didn't work because the Zero trait inherits from Add, and I'm not sure we want an additive structure on the Ranges.